AzureFunctionForSplunk/SecurityCenterLogs/SecurityLog.cs

```diff
@@ -0,0 +1,51 @@
+﻿//
+// AzureFunctionForSplunkVS
+//
+// Copyright (c) Microsoft Corporation
+//
+// All rights reserved. 
+//
+// MIT License
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy 
+// of this software and associated documentation files (the ""Software""), to deal 
+// in the Software without restriction, including without limitation the rights 
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell 
+// copies of the Software, and to permit persons to whom the Software is furnished 
+// to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in all 
+// copies or substantial portions of the Software.
+
+// THE SOFTWARE IS PROVIDED *AS IS*, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR 
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS 
+// FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR 
+// COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER 
+// IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION 
+// WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+//
+using AzureFunctionForSplunk.Common;
+using System.Collections.Generic;
+
+namespace AzureFunctionForSplunk.SecurityCenterLogs
+{
+    public class SecurityLog : AzMonMessage
+    {
+        public SecurityLog(dynamic message)
+        {
+            Message = message;
+
+            if (((IDictionary<string, object>)message).ContainsKey("AzureResourceId"))
+            {
+                ResourceId = message.AzureResourceId;
+            }
+            else
+            {
+                ResourceId = message.properties.resourceDetails.id;
```

Comment:
Can you post a raw example of a security center log record? The reason that a couple of the other log types look for both resourceId and resourceid is that over time the product team has published logs with slightly different spellings (case) of the thing. But it's not easy to find that sort of detail in any docs.